firmware/boron-ins-fsm/src/stateMachine.cpp

```diff
@@ -208,17 +212,19 @@ void state2_duration(){
     Log.warn("Door opened, session over, going to idle from state2_duration");
     publishStateTransition(2, 0, checkDoor.doorStatus, checkINS.iAverage);
     saveStateChange(2, 2);
+    hasDurationAlertBeenSent = 0; // session is over, so reset flag
     stateHandler = state0_idle;
-
   }
-  else if(millis() - state2_duration_timer >= state2_max_duration){
+  else if((millis() - state2_duration_timer >= state2_max_duration)
+          && !hasDurationAlertBeenSent){
 
     Log.warn("See duration alert, going from state2_duration to idle after alert publish");
     publishStateTransition(2, 0, checkDoor.doorStatus, checkINS.iAverage);
```

Comment:
This is no longer accurate because you're going to state 2 instead of idle.

Reply:
I updated this in my latest commit by removing the publish state transition. However, do you think that we should keep the saveStateChange() here? We publish an alert, but do not change states - please let me know what you think. I personally think we should remove or modify it.

Reply:
Good question. See earlier response about `saveStateChange`.